ts/sagas/services/servicesOptinSaga.ts

```diff
@@ -14,3 +14,15 @@ export function* askServicesOptin() {
 
   yield take(servicesOptinCompleted);
 }
+
+export function* askOldUsersServicesOptin() {
```

Comment:
imho we can easily forget what `old` means, maybe we can refer to these users as _alreadyLogged_?
The starup saga (`ts/sagas/startup.ts`) knows if the user has to login or if he/she is already logged.

`askServicesOptin` and `askOldUsersServicesOptin` are very similar, could this refactor be an improvement?

```typescript
/**
 * navigate to a screen where the user can set his/her services preference mode
 * if the user is not in onboarding (alreadyLogged = true) flow a thank you page will be shown at the end
 * @param alreadyLogged
 */
export function* servicePreferenceModeOptIn(alreadyLogged: true) {
  // TODO Check if the User already set this information
  yield put(
    NavigationActions.navigate({
      routeName: ROUTES.ONBOARDING_SERVICES_PREFERENCE,
      params: { alreadyLogged }
    })
  );
  if (alreadyLogged) {
    // remove the thank you page from the navigation stack
    yield put(navigationHistoryPop(1));
  }
  yield take(servicesOptinCompleted);
}
```

note: I added a comment on `navigationHistoryPop(1)`, could you check if it speaks the truth?

Reply:
Yes you're actually right, those two sagas were just raw implementation to integrate the preference request and thank you screens into the onboarding funnels.

Maybe we can ignore at the moment the refactoring and move it during the development by the releated [story](https://pagopa.atlassian.net/browse/IARS-25) what do you think?

Reply:
```suggestion
// TODO this saga need to be refactored as part of https://pagopa.atlassian.net/browse/IARS-25
export function* askOldUsersServicesOptin() {
```